core/sds-aspect-model-validator/src/main/java/io/openmanufacturing/sds/aspectmodel/validation/services/ViolationRustLikeFormatter.java

```diff
@@ -0,0 +1,212 @@
+ * information regarding authorship.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package io.openmanufacturing.sds.aspectmodel.validation.services;
+
+import org.apache.jena.rdf.model.Model;
+
+import io.openmanufacturing.sds.aspectmodel.resolver.parser.IRdfTextFormatter;
+import io.openmanufacturing.sds.aspectmodel.shacl.RustLikeFormatter;
+import io.openmanufacturing.sds.aspectmodel.shacl.fix.Fix;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.ClassTypeViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.ClosedViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.DatatypeViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.DisjointViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.EqualsViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.InvalidValueViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.JsConstraintViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.LanguageFromListViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.LessThanOrEqualsViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.LessThanViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MaxCountViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MaxExclusiveViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MaxInclusiveViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MaxLengthViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MinCountViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MinExclusiveViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MinInclusiveViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MinLengthViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.MissingTypeViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.NodeKindViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.NotViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.PatternViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.SparqlConstraintViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.UniqueLanguageViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.ValueFromListViolation;
+import io.openmanufacturing.sds.aspectmodel.shacl.violation.Violation;
+
+public class ViolationRustLikeFormatter extends ViolationFormatter {
+
+   private final RustLikeFormatter formatter;
+
+   private final Model rawModel;
+
+   public ViolationRustLikeFormatter() {
+      rawModel = null;
```

Comment:
If I understand correctly, if the ViolationRustLikeFormatter is created using this constructor, `null` will be passed to `RustLikeFormatter#constructDetailedMessage`. Is this intentional? If yes, either the model argument to the method should be made `Optional<Model>` or be annotated `javax.annotation.Nullable`.

Reply:
Yes, this is intentional. In many cases (especially unit tests) we do not have compound models (merged from several source files), so the rawModel does not need to be specified explicitly via this parameter, it can be deduced directly from the "highligh node".
Added the @Nullable annotation to the method parameter.